### PR TITLE
Jl/mmp 3048/caip multichain error handling cleanup

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Error cleanup

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26825?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/MetaMask-planning/issues/3048

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/multichain-api/scope/validation.ts [hunk 1]
```diff
@@ -36,21 +35,13 @@ export const isValidScope = (
 
   // These assume that the namespace has a notion of chainIds
   if (reference && scopes && scopes.length > 0) {
-    // TODO: Probably requires refactoring this helper a bit
-    // When a badly-formed request includes a chainId mismatched to scope
-    //   code = 5203
-    //   message = "Scope/chain mismatch"
-    // When a badly-formed request defines one chainId two ways
-    //  code = 5204
-    //  message = "ChainId defined in two different scopes"
```
Comment (jiexi) [inline app/scripts/lib/multichain-api/scope/validation.ts:45]: I don't think we need to throw these. Our current strategy has been to drop/ignore malformed ScopeObjects, and so these sort of errors don't really apply to us

File: app/scripts/lib/multichain-api/scope/assert.ts [hunk 2]
```diff
@@ -7,6 +7,64 @@ import {
...
+//   if (!areNotificationsValid) {
+//     throw new Error('Expected notifications to be non-empty strings')
+//   }
+
+//   // unexpected properties found on scopeObject
+//   if (Object.keys(restScopeObject).length !== 0) {
+//     throw new Error('Unexpected properties properties on scopeObject')
+//   }
+// };
+
```
Comment (jiexi) [inline app/scripts/lib/multichain-api/scope/assert.ts:67]: Can be removed if we decided that we indeed do not need to fire 5203 and 5204

File: app/scripts/lib/multichain-api/scope/validation.ts (hunk 1 above)
Comment (jiexi) [reply app/scripts/lib/multichain-api/scope/validation.ts:45]: I think we have landed on not firing these

**Comment by jiexi** [comment]
Unsure what to do here https://github.com/MetaMask/metamask-extension/pull/25665/files#diff-617983b14a95b3b1326a1df8249459b95ebbc3824141bc08dda7437137231755R30

**Review by shanejonas — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
Looks right to me!!!

**Pull request merged**